projects/core/src/cart/event/cart-event.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { StateEventService } from '../../state/event/state-event.service';
+import { CartActions } from '../store/actions';
+import { CartEvents } from './cart-event.model';
+
+@Injectable({ providedIn: 'root' })
+export class CartEventService {
+  constructor(protected stateEventService: StateEventService) {
+    this.register();
+  }
+
+  /**
+   * Registers event sources
+   */
+  protected register() {
+    this.stateEventService
+      .register({
+        action: CartActions.CART_ADD_ENTRY,
+        event: CartEvents.AddCartEntry,
+      })
+      .register({
```

Comment:
To reduce the need to write again `this.stateEventService`, I made `register()` chainable.

On the other hand we loose the ability to get a teardown function. But since action-to-event mapping is permanent, we don't need to tear down.